src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -787,20 +793,29 @@ export class SavedObjectsRepository {
     }
 
     const esOptions = {
-      index: this.getIndicesForTypes(allowedTypes),
+      // If `pit` is provided, we drop the `index` and `preference` as those are already
+      // associated with the PIT in ES, and will otherwise return a 400.
+      ...(pit
+        ? {}
+        : {
+            index: this.getIndicesForTypes(allowedTypes),
+            preference,
+          }),
+      // If `searchAfter` is provided, we drop `from` as it will not be used for pagination.
+      ...(searchAfter ? {} : { from: perPage * (page - 1) }),
```

Comment:
There are 3 ways we could have done this:
1. Implicitly drop unneeded options (what I've done here)
2. Explicitly return an error if the wrong combo of options is provided
3. Make the typings more complex and attempt to enforce that only the accepted combos of options are provided

As I was writing the exports code, (1) made things much easier to work with, but open to discussing the other approaches if anyone feels strongly about it.

Reply:
In general I also prefer being explicit. Sometimes it creates confusion where you think you're using the API in a certain way because of a paramater without realizing that it has no effect.

Reply:
Upon revisiting this, I realized that the `from` value is derived from the `perPage` setting, and while `index` must be excluded, the `types` are still necessary for the authorization checks.

So in the end we are only really talking about whether we implicitly allow passing both `preference` and `pit`. To be conservative, I went ahead and added an explicit check for that combination, and `find` will now throw if you attempt to provide both.